WooCommerce/Classes/ViewRelated/Products/Edit Product/Shipping Settings/ProductShippingSettingsViewModel.swift

```diff
@@ -242,5 +242,6 @@ private extension ProductSubscription {
     var supportsOneTimeShipping: Bool {
         (trialLength.isEmpty || trialLength == "0")
         && (paymentSyncDate.isEmpty || paymentSyncDate == "0")
+        && (paymentSyncMonth.isEmpty || paymentSyncMonth == "0")
```

Comment:
I'm not sure about this check - my product (Nintendo online subscription) doesn't have sync renewal set up, but it has `paymentSyncMonth` as "1". I don't know what that means.

At the same time, I'm seeing `paymentSyncMonth` for that same product being "" instead of "1". Could you check if it's a problem with the decoding or the caching?

Reply:
Good catch about this. For this product the `day` field has Int value instead of String. 

1. Added custom decoding to handle String and Int types in https://github.com/woocommerce/woocommerce-ios/pull/11312/commits/91a9ef15497c04fb09a46937a7d2b5ed937a577a and https://github.com/woocommerce/woocommerce-ios/pull/11312/commits/5ac6e386d1340b0eb415f334d526705bedd3b322
1. Stopped checking for month value in https://github.com/woocommerce/woocommerce-ios/pull/11312/commits/33d26caf68968badfdf0b91908596cb2dc6f9f54. Internal - p1701256402451759/1701252118.837079-slack-C03L1NF1EA3